core/src/bmi.jl

```diff
@@ -432,13 +432,15 @@ function set_control_params!(p::Parameters, node_id::Int, control_state::String)
 
     for (field, value) in zip(keys(new_state), new_state)
         if !ismissing(value)
-            getfield(node, field)[idx] = value
+            vec = preallocation_dispatch(getfield(node, field), 0)
+            vec[idx] = value
         end
     end
 end
 
 "Copy the current flow to the SavedValues"
-save_flow(u, t, integrator) = copy(nonzeros(integrator.p.connectivity.flow))
+save_flow(u, t, integrator) =
+    copy(nonzeros(preallocation_dispatch(integrator.p.connectivity.flow, u)))
```

Comment:
If it doesn't fit on a single line I generally prefer to switch to the long function syntax:

```
function f()
    ...
end
```